packages/babel-helpers/src/helpers/applyDecs.js

```diff
@@ -703,7 +703,7 @@ function applyClassDecs(ret, targetClass, metadataMap, classDecs) {
 
   initializeClass(Class);
  */
-export default function applyDecs(targetClass, memberDecs, classDecs) {
+export default function applyDecs(targetClass, classDecs, memberDecs) {
```

Comment:
Technically this is a breaking change. But it will only affect `@babel/helper 7.17.0`. Since 7.17.0 was just released two days ago, I think we could just bump the minimum version helpers requirements to `7.17.2`. Alternatively we can add `applyDecs2` like we did for `objectRest` before.

Reply:
Another approach is to maintain current interface and memoize evaluated decorator expressions. I think we must do so because decorator lists, as well as computed property keys are evaluated outside the class scope, e.g.

```js
function decFactory(name) { console.log(name); return x => x }
class B {
  static {
   class C {
     @decFactory(this.name) p;
   }
  }
}
```
The expected behaviour is to log `"B"`, currently the transpiled result logs `"C"` because the `decFactory()` is evaluated within a static block of `C`.